railties/lib/rails/generators/rails/authentication/templates/app/channels/application_cable/connection.rb.tt

```diff
@@ -0,0 +1,17 @@
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+    identified_by :current_user
+
+    def connect
+      set_current_user || reject_unauthorized_connection
+    end
+
+    private
+      def set_current_user
+        if session = Session.find_by(id: cookies.signed[:session_id])
+          self.current_user = session.user
+        end
```

Comment:
There is currently an open PR (https://github.com/rails/rails/pull/53220) that adds a conditional to the `Authentication#find_session_by_cookie` to guard against issuing an SQL query when the cookie is not set. Could this code include a similar check?

```suggestion
        if (id = cookies.signed[:session_id]) && (session = Session.find_by(id: id)
          self.current_user = session.user
        end
```

Reply:
I don't think that makes sense here given that this connection is assuming that everyone needs to be authenticated or be rejected. Therefore the rejection is essentially an exceptional state. You shouldn't be trying to establish a connection unless you believe it would succeed. So this complicates the code for a condition that isn't worth optimizing for.